src/createRestyleFunction.ts

```diff
@@ -73,26 +72,28 @@ const getValue = <Theme extends BaseTheme>(
   return val;
 };
 
-const createRestyleFunction = ({
+const createRestyleFunction = <
+  TProps = Record<string, unknown>,
+  Theme extends BaseTheme = BaseTheme,
+  P extends keyof TProps = keyof TProps,
```

Comment:
Is there a valid case for when you should be able to override P and K here? If not, I would prefer to just using `keyof TProps` and `keyof Theme` inline within the function.

Reply:
Its not so much about overriding as it is about having a strongly typed return type so that the exact property and themeKey the user specified is preserved and carried through to anything consuming that restyle function.

Reply:
Ah right, got it